tuda_workspace_scripts/discovery.py

```diff
@@ -0,0 +1,126 @@
+from .workspace import get_workspace_root
+from robots import DiscoveryServer, load_robots
+from tuda_workspace_scripts import print_warn
+from ros2cli.node.daemon import is_daemon_running, spawn_daemon, shutdown_daemon
+import os
+import xml.etree.ElementTree as ET
+
+
+# dedicated place for temporary files
+tmp_directory = "/tmp/tuda_wss"
```

Comment:
Constants are usually SCREAMING_SNAKE_CASE.
Also this file was not black formatted.